docs/x509-auth.md

```diff
@@ -0,0 +1,148 @@
+
+1. To install the MongoDB Community Kubernetes Operator, copy and paste 
+   the following command and replace the `<namespace>` variable with the 
+   namespace:
+
+   ```
+   helm upgrade --install community-operator mongodb/community-operator \
+   --namespace <namespace> --set namespace=<namespace> --create-namespace \
+   --set resource.tls.useCertManager=true --set resource.tls.enabled=true \
+   --set resource.tls.useX509=true
+   ```
+
+1. To deploy the MongoDB Community Kubernetes Operator, copy and paste 
+   the following command and replaces the `<namespace>` variable with the 
+   namespace:
+
+   ```
+   helm upgrade --install community-operator mongodb/community-operator \
+   --namespace <namespace> --set namespace=<namespace> --create-namespace \
+   --set resource.tls.useCertManager=true --set resource.tls.enabled=true \
+   --set resource.tls.useX509=true --set resource.tls.sampleX509User=true \
+   --set createResource=true
+   ```
+
+## Use `kubectl` to Enable X.509 Authentication
+
+You can use Helm to install and deploy the MongoDB Community Kubernetes 
+Operator with X.509 Authentication enabled for the MongoDB Agent and 
+client.
+
+1. To install the MongoDB Community Kubernetes Operator, see 
+   [Install the Operator using kubectl](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/install-upgrade.md#install-the-operator-using-kubectl)
+
+1. To create a CA, ConfigMap, secrets, issuer, and certificate, see 
+   [Enable External Access to a MongoDB Deployment](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/external_access.md)
+
+1. Create a YAML file for the  MongoDB Agent certificate. For an example, 
+   see [agent-certificate.yaml](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/config/samples/external_access/agent-certificate.yaml).
+
+   **Note:**
+
+   - For the `spec.issuerRef.name` parameter, specify the 
+     `cert-manager` issuer that you created previously.
+   - For the `spec.secretName` parameter, specify the same 
+     value as the `spec.security.authentication.agentCertificateSecretRef` 
+     parameter in your resource. This secret should contain a signed 
+     X.509 certificate and a private key for the MongoDB agent.
+
+1. To apply the file, copy and paste the following command and replace 
+   the `<agent-certificate>` variable with the name of your MongoDB Agent 
+   certificate and the `<namespace>` variable with the namespace:
+
+   ```
+   kubectl apply -f <agent-certificate>.yaml --namespace <namespace>
+   ```
+
+1. Create a YAML file for your resource. For an example, see 
+   [mongodb.com_v1_mongodbcommunity_x509.yaml](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/config/samples/mongodb.com_v1_mongodbcommunity_x509.yaml).
+
+   **Note:**
+
+   - For the `spec.security.tls.certificateKeySecretRef.name` parameter,
```

Comment:
You can also instead link to this doc explaining the structure and more: https://github.com/mongodb/mongodb-kubernetes-operator/blob/60da4dc17911ba9200a1365547c7c44cecd0bf54/config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml#L414-L424 

its a certificate that the server not generates but uses I think